Update EUI to 110.0.0

mgadewoll opened the pull request:
## Dependency updates

- `@elastic/eui`: `v109.2.0` ⏩ `v110.0.0`

---

## Changes

- Updates usages of the `message` prop on `EuiInMemoryTable` to `noItemsMessage`
- Updates usages of `EuiSearchBar`'s `onChange` prop to check for the `query` argument, as it may be `null` now
- Removes obsolete `color` props on implementations of `EuiPageTemplate` and `EuiPageTemplate.Header` (`color` prop wasn't applied before either)

## Package updates

### `@elastic/eui` v110.0.0

- Updated `EuiSuperDatePicker` to expose plain text `utcOffset` and `timeZoneName` in `timeZoneDisplayProps.customRender` render function ([#9245](https://github.com/elastic/eui/pull/9245))
- Updated `EuiHeaderLogo` to add a hover style ([#9240](https://github.com/elastic/eui/pull/9240))
- Made `EuiQuickSelectPanel` available for importing from the `@elastic/eui` package ([#9239](https://github.com/elastic/eui/pull/9239))
- Updated `EuiAvatar` to support Emoji Sequence ("advanced") in the `initials` prop ([#9227](https://github.com/elastic/eui/pull/9227))
- Added `color` prop on `EuiPageHeader` to support `transparent` (default) and `plain` backgrounds. ([#9220](https://github.com/elastic/eui/pull/9220))
- Added `color` prop on `EuiPage` to support `transparent` (default) and `plain` backgrounds. ([#9220](https://github.com/elastic/eui/pull/9220))
- Updated `EuiPageTemplate` to ensure `panelled=true` renders a `EuiPageHeader` with a plain background by default ([#9220](https://github.com/elastic/eui/pull/9220))
- Removed the default background style on `EuiPageTemplate`'s outer wrapper ([#9220](https://github.com/elastic/eui/pull/9220))

**Bug fixes**

- Fixed icon size in `EuiSuperDatePicker`'s time window buttons when `compressed={true}` ([#9245](https://github.com/elastic/eui/pull/9245))
- Fixed `EuiIcon` visibility issue with `logoElastic` when `color` is set to `text` or `ghost` in light mode ([#9247](https://github.com/elastic/eui/pull/9247))
- Fixed `EuiInMemoryTable` support for controlled search for plain text (when `searchFormat="text"`) by properly handling `search.query` and `search.defaulQuery` ([#9142](https://github.com/elastic/eui/pull/9142))

**Breaking changes**

- Removed deprecated `message` prop from `EuiInMemoryTable`, use `noItemsMessage` instead ([#9234](https://github.com/elastic/eui/pull/9234))

**Accessibility**

- Improved the accessibility of input fields in the popover of `EuiSuperDatePicker` by properly labeling them ([#9239](https://github.com/elastic/eui/pull/9239))
- Improved the accessibility of `EuiSelectable` by removing empty `aria-activedescendant` attribute when no option is active to ensure the search input is perceivable by screen readers ([#9223](https://github.com/elastic/eui/pull/9223))

weronikaolejniczak committed:
chore(eui): use noItemsMessage prop in EuiInMemoryTable

Commit:
Update test to reflect role change of search input from "searchbox" to "combobox" in DataView list component

mgadewoll committed:
chore: update eui packages

mgadewoll committed:
refactor: update message prop

mgadewoll committed:
refactor: remove obsolete color prop

mgadewoll committed:
refactor: check for available query arg

- EUI updated the type and the query might be null for text searches

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

mgadewoll committed:
test: use noItemsMessage prop

mgadewoll committed:
test: update snapshots

mgadewoll committed:
Merge branch 'main' into eui/v110.0.0

mgadewoll committed:
Merge branch 'main' into eui/v110.0.0

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

rmyz reviewed (approved):
obs presentation changes LGTM

sabarasaba reviewed (approved):
kibana management changes lgtm

jillguyonnet reviewed (approved):
Fleet changes LGTM 🚀

elasticmachine commented:
Pinging @elastic/obs-presentation-team (Team:obs-presentation)

PhilippeOberti reviewed (approved):
Code review only, LGTM for the @elastic/security-threat-hunting-investigations team

peteharverson reviewed (approved):
ML changes LGTM (tested the 3 tables with `noItemsMessage` changes).

consulthys reviewed (approved):
LGT Stack Monitoring
Thanks for the updates

rylnd reviewed (approved):
Detection Engine changes LGTM.

ThomThomson reviewed (approved):
Presentation team changes LGTM!

TinaHeiligers reviewed (approved):
saved objects management snapshot update LGTM

mashhurs reviewed (approved):
Logstash pipelines view LGTM!

jeramysoucy reviewed (approved):
LGTM - code review and manually checked each modified page

elasticmachine committed:
Merge branch 'main' into eui/v110.0.0

mgadewoll committed:
refactor: update profile header styles

- prevents background color differences due to the spacing resulting from the max-width style

mgadewoll commented on x-pack/platform/plugins/shared/security/public/account_management/user_profile/user_profile.tsx:
```diff
@@ -78,9 +78,12 @@ const formRowCSS = css`
-  margin: auto;
   border-bottom: none;
+
+  .euiPageHeaderContent {
```
ℹ️ This prevents a visual issue where different background colors are visible in the Profile page header (Classic Kibana only). This is due to the background color change on the `EuiPage`/`EuiPageTemplate`.

| before | after |
|---|---|
| <img width="1280" height="232" alt="Screenshot 2025-12-05 at 11 11 26" src="https://github.com/user-attachments/assets/9fc9aac6-8faa-465a-9c14-6a3d6879f974" /> | <img width="1282" height="230" alt="Screenshot 2025-12-05 at 11 14 03" src="https://github.com/user-attachments/assets/3f32b266-2931-468d-b00b-dbdde2b3c8e8" /> |

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369419)
* Commit: 9e620055b0a9d32a0fa5c06d8d9639a3475bba3c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-244866)

### Failed CI Steps
* [Post-Build](https://buildkite.com/elastic/kibana-pull-request/builds/369419#019aee04-bf92-406b-a520-d12a31b4a425)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: no builds found for mergeBase sha [cb654f2263f6e9ab0024b4930850189e936faa69]

### History
* :broken_heart: [Build #369388](https://buildkite.com/elastic/kibana-pull-request/builds/369388) failed de874d9c3b793a26c2e29fb40a9df11c538d97db
* :yellow_heart: [Build #367950](https://buildkite.com/elastic/kibana-pull-request/builds/367950) was flaky 5c00e7fc3407f03dfebced13595f358ccee45aef
* :broken_heart: [Build #367862](https://buildkite.com/elastic/kibana-pull-request/builds/367862) failed 3de7124881934ec6bab01e98f07b3e4c5ac881dd
* :broken_heart: [Build #367605](https://buildkite.com/elastic/kibana-pull-request/builds/367605) failed 710a7ac732a8564fd3bec110623385ef7f6e7013
* :broken_heart: [Build #367520](https://buildkite.com/elastic/kibana-pull-request/builds/367520) failed 9318d596e27d216fcd25d7dd0bb0bd23e037ec2b

cc @mgadewoll

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369419","number":369419,"commit":"9e620055b0a9d32a0fa5c06d8d9639a3475bba3c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369388","number":369388,"commit":"de874d9c3b793a26c2e29fb40a9df11c538d97db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367950","number":367950,"commit":"5c00e7fc3407f03dfebced13595f358ccee45aef"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367862","number":367862,"commit":"3de7124881934ec6bab01e98f07b3e4c5ac881dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367605","number":367605,"commit":"710a7ac732a8564fd3bec110623385ef7f6e7013"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367520","number":367520,"commit":"9318d596e27d216fcd25d7dd0bb0bd23e037ec2b"}],"number":369419}
buildkite-pr-comment-->

elasticmachine committed:
Merge branch 'main' into eui/v110.0.0

shahzad31 reviewed (approved):
LGTM !!
